blocks/editable/index.js

```diff
@@ -51,6 +51,12 @@ function createTinyMCEElement( type, props, ...children ) {
 	);
 }
 
+function isLinkBoundary( fragment ) {
+	return fragment.childNodes && fragment.childNodes.length === 1 &&
+		fragment.childNodes[ 0 ].nodeName === 'A' && fragment.childNodes[ 0 ].text.length === 1 &&
+		fragment.childNodes[ 0 ].text[ 0 ] === '\uFEFF';
```

Comment:
It's fine for now, but maybe this character could be included in a constant somewhere in TinyMCE to avoid issues if it's changed.